crates/zed/src/zed/open_listener.rs

```diff
@@ -261,7 +261,12 @@ pub async fn handle_cli_connection(
                 wait,
                 open_new_workspace,
                 env,
+                user_data_dir,
             } => {
+                if let Some(dir) = user_data_dir {
+                    paths::set_custom_data_dir(&dir);
```

Comment:
I don't think we need this, as you're already passing it via CLI arguments and set_custom_data_dir() should not be called after `init_paths`. In fact, can we make `set_custom_data_dir()` panic if you call it after calling `init_paths`?

Reply:
updated